mobify/progressive-web-scaffold
# WEB-1236: Optimize script loading

The latest in the Merlin's Performance Optimization changes committed upstream from work done on other projects!

 **JIRA**: [WEB-1243](https://mobify.atlassian.net/browse/WEB-1243)

## Changes
- Load scripts asynchronously in the same manner as [we did here](https://github.com/mobify/lancome-progressive/pull/849)
- Impact: not nearly as dramatic as in the above example. In both before/after, the site is blocked by `vendor.js` anyway.

| Before | After |
| ------- | ----- |
| _Notice the red box: that is the time waited before main.js starts to download due to lack of async loading_ ![screen shot 2017-03-09 at 3 54 07 pm](https://cloud.githubusercontent.com/assets/734535/23778260/2ad7f7ca-04ee-11e7-84d0-40a2c325e970.png) | _Notice how all scripts now load at the same time_ ![screen shot 2017-03-09 at 5 03 03 pm](https://cloud.githubusercontent.com/assets/734535/23778274/3aa0b156-04ee-11e7-83cb-886244318ec4.png) |

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Verify that all four scripts load at the same time: `jquery.min.js`, `capture-latest.min.js`, `main.js`, and `vendor.js`
